tasks/selinux_load_module.yml

```diff
@@ -0,0 +1,55 @@
+- name: Install module
+  block:
+    - name: Create temporary directory
+      tempfile:
+        state: directory
+        prefix: linux_system_role.selinux
+      register: tempdir
+
+    - name: Copy module file "{{ item.path }}"
+      copy:
+        dest: "{{ tempdir.path }}"
+        src: "{{ item.path }}"
+
+    - name: Install module with priority
+      command: "semodule -n -s {{ store }} -X {{ priority }} -i {{ tempdir.path }}/{{ item.path | basename }}"
+      when: priority != "0" and selinux_priorities
+
+    - name: Install module
+      command: "semodule -n -s {{ store }} -i {{ tempdir.path }}/{{ item.path | basename }}"
+      when: priority == "0" or not selinux_priorities
+
+  when:
+    state == "enabled" and item.path is defined
+  always:
+    - name: Clean temporary directory "{{ tempdir.path }}"
+      file:
+        path: "{{ tempdir.path }}"
+        state: "absent"
+
+- name: Enable module with priority
+  command: "semodule -n -s {{ store }} -X {{ priority }} -e {{ item.name }}"
+  when: state == "enabled" and item.path is not defined and current_state == "disabled" and (priority != "0" and selinux_priorities)
+
+- name: Disable module with priority
+  command: "semodule -n -s {{ store }} -X {{ priority }} -d {{ item.name }}"
+  when: state == "disabled" and current_state == "enabled" and (priority != "0" and selinux_priorities)
+
+- name: Remove module with priority
+  command: "semodule -n -s {{ store }} -X {{ priority }} -r {{ item.name }}"
+  when: state == "absent" and current_state != "absent" and (priority != "0" and selinux_priorities)
+
+- name: Enable module
+  command: "semodule -n -s {{ store }} -e {{ item.name }}"
+  when: state == "enabled" and item.path is not defined and (priority == "0" or not selinux_priorities)
+
+- name: Disable module
+  command: "semodule -n -s {{ store }} -d {{ item.name }}"
+  when: state == "disabled" and (priority == "0" or not selinux_priorities)
+
+- name: Remove module
+  command: "semodule -n -s {{ store }} -r {{ item.name }}"
+  when: state == "absent" and (priority == "0" or not selinux_priorities)
+
+- name: Refresh SELinux modules facts
```

Comment:
If you need to reload the policy i.e. make the handler run, you can use the `meta` module:
```
  - name: Flush handlers
    meta: flush_handlers
```
but this will flush all of the pending handlers in the entire play (@pcahyna is this correct?)

If we can't use a handler because it would trigger every pending handler in the entire play, then you'll have to use a `register` on each task that modifies the remote state e.g.
```

    - name: Install module with priority
      command: "semodule -n -s {{ store }} -X {{ priority }} -i {{ tempdir.path }}/{{ item.path | basename }}"
      when: priority != "0" and selinux_priorities
      register: __selinux_register_install_pri

    - name: Install module
      command: "semodule -n -s {{ store }} -i {{ tempdir.path }}/{{ item.path | basename }}"
      when: priority == "0" or not selinux_priorities
      register: __selinux_register_install
...
    - name: Set __selinux_changed flag
      set_fact:
        __selinux_changed: true
      when: |
        (__selinux_register_install_pri is defined and __selinux_register_install_pri is changed)
        or (__selinux_register_install is defined and __selinux_register_install is changed)
        or ....
```
then in main.yml
```
- name: Reload SELinux policy
  command: semodule -R
  when:
    - ansible_selinux.status != "disabled"
    - __selinux_changed | d(false)

- name: Reset the changed flag
  set_fact:
    __selinux_changed: false
```
I think this is the only way to make this role idempotent

Reply:
https://github.com/linux-system-roles/selinux/pull/56/commits/7f2337be2a174bc9f03db35d7998e91af38bbf76